voice/src/main/java/discord4j/voice/AudioProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * This file is part of Discord4J.
+ *
+ * Discord4J is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Discord4J is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Discord4J.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package discord4j.voice;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Used to send audio.
+ * <p>
+ * The provider uses a shared buffer. Keep this in mind when implementing.
+ *
+ * @see #provide()
+ */
+public abstract class AudioProvider {
+
+    public static final int DEFAULT_BUFFER_SIZE = 2048;
+    public static final AudioProvider NO_OP = new AudioProvider(ByteBuffer.allocate(0)) {
+        @Override
+        public boolean provide() {
+            return false;
+        }
+    };
+
+    private final ByteBuffer buffer;
```

Comment:
In general, I think we should prefer netty's `ByteBuf`

Reply:
It being nio ByteBuffer makes it much easier to interact with other libraries (who understandably don't want to pull netty just for ByteBuf). For example, the current LavaPlayer [demo](https://github.com/Discord4J/Discord4J/blob/feature%2Fvoice/core/src/test/java/discord4j/core/ExampleVoiceBot.java#L100) relies on ByteBuffer.